### PR TITLE
Setup unit test

### PR DESCRIPTION
**Pull request by yehezkielgunawan**
# Description
Add base setup for unit testing. @lauslim12 please review this in your free time, no need to be rush

# Linked Issues
Close [https://github.com/yehezkielgunawan/wa-helper/issues/54](https://github.com/yehezkielgunawan/wa-helper/issues/54)

# Preview or Screenshot
-

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "start": "next start",
     "lint": "next lint",
     "prepare": "husky install",
-    "release": "standard-version"
+    "release": "standard-version",
+    "test": "jest --watch"
```
Comment (lauslim12) [inline package.json:12]: I think it's better if you add another command just to run all of tests in CI environment. If you're only doing `jest`, then that command usually only runs the test for the files which have changed since the last commit (which may or may not be what you want, sometimes you have to press another key after the first test has finished running to run all of the tests).

```suggestion
    "test": "jest --watch"
    "test:ci": "jest --ci --collect-coverage --verbose"
```

File: src/__test__/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// __tests__/index.test.jsx
```
Comment (lauslim12) [inline src/__test__/index.test.tsx:1]: Is this line necessary? 😅

File: src/__test__/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import { render } from "@testing-library/react";
+
+import "@testing-library/jest-dom";
+import Home from "@/pages/index";
+
+describe("Render Main Page Successfully", () => {
+  it("Renders Home Page Successfully", () => {
+    render(<Home countryCodes={[]} />);
+  });
+});
```
Comment (lauslim12) [inline src/__test__/index.test.tsx:12]: This is correct, but I believe the description is kind of redundant. I think it's better to reword it to be something like:

```suggestion
describe("<Home />", () => {
  it("renders home page successfully", () => {
    render(<Home countryCodes={[]} />);
  });
});
```

We can use either proper case or all in lowercase (I think it's Jest convention, but it depends on the project as well). It depends on your choice.

**Review by lauslim12 — APPROVED** [review]
I have taken a look and have some small comments. Well done!

File: package.json (hunk 1 above)
Comment (yehezkielgunawan) [reply package.json:12]: Oh alright, after setup the unit test, I'll also setup the CI/CD, nice sugesstion

File: src/__test__/index.test.tsx (hunk 2 above)
Comment (yehezkielgunawan) [reply src/__test__/index.test.tsx:1]: Will delete it soon in next commit😅
